doc/modules/ROOT/pages/index.adoc

```diff
@@ -1,46 +1,336 @@
+include::{templates-url}/link-to-repo.adoc[]
 
-// Use this relative url if you are going to publish the guide on the guides site.
-// Note that this url will not work locally and raise asciidoctor errors.
-// So, complete the guide with the above url and set the below one just before 
-// publishing on the guides site.
-//
-// :templates-url: templates:ROOT:page$/
+== What You’ll Learn
 
-= Title
+In this guide, you will deploy two Spring Boot applications that use Hazelcast for HTTP session store.
 
-// Content entered directly below the header but before the first section heading is called the preamble.
+Spring Session manages users' session information and supports clustered sessions rather than an application
+container-specific solution only. That is, more than one application instance can use the same store for user
+sessions. You can find more details on
+https://docs.spring.io/spring-session/docs/current/reference/html5/#introduction[Spring Session Documentation].
 
-include::{templates-url}/link-to-repo.adoc[]
+== Prerequisites
 
-== What You’ll Learn
+- ~10 minutes
+- JDK 1.8+
+- Apache Maven 3.2+
 
-// Define the problem and tell reader what he/she will learn from this guide.
+== Enabling HazelcastHttpSession
 
-== Prerequisites
+Before using Hazelcast session repository, let's add the required dependencies:
+
+.pom.xml
+[source, xml]
+----
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast</artifactId>
+    <version>${hazelcast.version}</version>
+</dependency>
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-web</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.session</groupId>
+    <artifactId>spring-session-core</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.session</groupId>
+    <artifactId>spring-session-hazelcast</artifactId>
+</dependency>
+----
+
+Now configure the app to use Hazelcast as HTTP session store:
+
+.SessionConfiguration.java
+[source, java]
+----
+@Configuration
+@EnableHazelcastHttpSession
+class SessionConfiguration {
+    // ...
+}
+----
+
+Alternatively, you may extend `HazelcastHttpSessionConfiguration` to have more advanced configurations
+rather than using annotation based configuration:
+
+[source, java]
+----
+@Configuration
+class SessionConfiguration extends HazelcastHttpSessionConfiguration {
+    // ...
+}
+----
+
+== Creating Hazelcast Instance Bean
+
+After enabling Hazelcast HTTP Session, you need to provide a Hazelcast instance to be used by
+the session repository. This instance can be either a Hazelcast client or an embedded Hazelcast instance.
+You can see the details of client and embedded modes https://docs.hazelcast.org/docs/latest/manual/html-single/#hazelcast-topology[here] on the documentation.
+
+Notice some `Hazelcast4` prefixes in the class names. If you use an older Hazelcast version (3.x), you
+need to drop these `"4"` s:
```

Comment:
Can it be good to have an example of Hazelcast 3.x? People might be confused about what we mean by dropping "4"s.